### PR TITLE
Support for accessing nested records in labels

### PR DESCRIPTION
**Pull request by alexfridlyand**
Added support to access nested records in labels via record accessor plugin helper.

### CODE REVIEW

**Comment by repeatedly** [comment]
Could you update README with nest field example for `<labels>`?

**Comment by alexfridlyand** [comment]
Updated README

**Comment by repeatedly** [comment]
Did you test this patch?
I tested this patch but it doesn't work.

I sent `{"foo":100,"bar":"hello","nest":{"key":"value"}}` event for following configuration but msg is empty.

```
<filter test.**>
  @type prometheus
  <metric>
    name message_foo_counter
    type counter
    desc The total number of foo in message.
    key foo
  </metric>
  <labels>
    tag ${tag}
    msg ${$.nest.key}
  </labels>
</filter>
```

**Comment by repeatedly** [comment]
Hmm... you expect raw `$.foo.bar` for configuration, not `${$.foo.bar}`?

**Comment by alexfridlyand** [comment]
Yes, i did some tests with kubernetes nested metadata and as per README additions (https://docs.fluentd.org/v1.0/articles/api-plugin-helper-record_accessor#syntax) such syntax worked for me. It's not like i insist on some syntax form, i just used the plugin provided and tested with existing test suite + some new nested records.

**Comment by repeatedly** [comment]
I see.

@kazegusuri How about this? Currently, we use `${foo}` for `foo` field. Is it ok for `$.foo.bar`, not `${$.foo.bar}`, for nested field access?

**Comment by kazegusuri** [comment]
sorry for no response.

I want to respect the official syntax if the plugin is a first plugin. So I'm okay for the syntax like `$.foo.bar`.
It will be complicated since there are `${foo}` and `$.foo` syntax for accessing record values. I think it is better the current syntax is only for built-in/reserved values and deprecated for record values.

File: lib/fluent/plugin/prometheus.rb [hunk 1]
```diff
@@ -101,7 +102,8 @@ def initialize(element, registry, labels)
         def labels(record, expander, placeholders)
           label = {}
           @base_labels.each do |k, v|
-            label[k] = expander.expand(v, placeholders)
+            accessor_value = v.call(record)
+            label[k] = expander.expand(accessor_value.nil? ? v.keys : accessor_value, placeholders)
```
Comment (kazegusuri) [inline lib/fluent/plugin/prometheus.rb]: This allows to use both RecordAccessor and RecordTransformer. IMO it's enough to use either one.
I want a value to be expanded by RecordTransfermer only when the RecordAccessor is applied.
Probably the code would be like this?
```
             label[k] = accessor_value.nil? ?  expander.expand(v.keys, placeholders) : accessor_value 
```

Comment (alexfridlyand) [reply lib/fluent/plugin/prometheus.rb]: Yes, that's better. Pushed changes and tested with nested record + original test suite

**Comment by kazegusuri** [comment]
please squash commits and the I will merge, thanks!

**Comment by alexfridlyand** [comment]
Just squashed (hopefully with the convention you support). Glad to collaborate, thanks.

**Comment by kazegusuri** [comment]
thank you!

**Comment by alexfridlyand** [comment]
@kazegusuri Do you plan to release it soon? I would like to reuse new functionality (much needed). Thanks

**Pull request merged**